### DISCUSSION TITLE
Use of useLayoutEffect causes errors with SSR

### DISCUSSION THREAD

**Discussion by rsmithclearspeed** [pmndrs/valtio#955] [Bug report] [answered]
### Bug Description

I'm currently incorporating valtio in a Remix project utilizing SSR. When the server rendering runs, I receive the error below. I was able to resolve this by patching the package to use useEffect instead of useLayoutEffect during the server side render ([like this](https://github.com/react-component/overflow/issues/6#issuecomment-990105776)). Should a patch like this be incorporated into the library?

`Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.`

### Reproduction Link

https://stackblitz.com/edit/remix-run-remix-ipk4mq?file=app%2Froutes%2F_index.tsx

**Comment by dai-shi** [comment, marked as answer]
Hi, thanks for reporting.
To be precise, it's a false-positive warning. React 19 won't have it. But it's annoying with React 18.
See https://github.com/pmndrs/valtio/discussions/652#discussioncomment-10502517 and our suggestion is #944.

**Comment by rsmithclearspeed** [comment]
Ah, yes I see that in the docs now. Thanks!
